phpBB/install/convert/controller/convertor.php

```diff
@@ -0,0 +1,857 @@
+			$this->iohandler->send_response();
+		}
+		else
+		{
+			$this->render_settings_form($error);
+		}
+	}
+
+	/**
+	 * Renders settings form
+	 *
+	 * @param array	$error	Array of errors
+	 */
+	public function render_settings_form($error = array())
+	{
+		foreach ($error as $msg)
+		{
+			$this->iohandler->add_error_message($msg);
+		}
+
+		$dbms_options = array();
+		foreach ($this->db_helper->get_available_dbms() as $dbms_key => $dbms_array)
+		{
+			$dbms_options[] = array(
+				'value'		=> $dbms_key,
+				'label'		=> 'DB_OPTION_' . strtoupper($dbms_key),
+			);
+		}
+
+		$form_title = 'SPECIFY_OPTIONS';
+		$form_data = array(
+			'src_dbms'	=> array(
+				'label'		=> 'DBMS',
+				'type'		=> 'select',
+				'options'	=> $dbms_options,
+			),
+			'src_dbhost' => array(
+				'label'			=> 'DB_HOST',
+				'description'	=> 'DB_HOST_EXPLAIN',
+				'type'			=> 'text',
+			),
+			'src_dbport' => array(
+				'label'			=> 'DB_PORT',
+				'description'	=> 'DB_PORT_EXPLAIN',
+				'type'			=> 'text',
+			),
+			'src_dbname' => array(
+				'label'		=> 'DB_NAME',
+				'type'		=> 'text',
+			),
+			'src_dbuser' => array(
+				'label'		=> 'DB_USERNAME',
+				'type'		=> 'text',
+			),
+			'src_dbpasswd' => array(
+				'label'	=> 'DB_PASSWORD',
+				'type'	=> 'password',
+			),
+			'src_table_prefix' => array(
+				'label'			=> 'TABLE_PREFIX',
+				'description'	=> 'TABLE_PREFIX_EXPLAIN',
+				'type'			=> 'text',
+			),
+			'forum_path' => array(
+				'label'			=> 'FORUM_PATH',
+				'description'	=> 'FORUM_PATH_EXPLAIN',
+				'type'			=> 'text',
+			),
+			'refresh' => array(
+				'label'			=> 'REFRESH_PAGE',
+				'description'	=> 'REFRESH_PAGE_EXPLAIN',
+				'type'			=> 'radio',
+				'options'		=> array(
+					array(
+						'value'		=> 0,
+						'label'		=> 'NO',
+						'selected'	=> true,
+					),
+					array(
+						'value'		=> 1,
+						'label'		=> 'YES',
+						'selected'	=> false,
+					),
+				),
+			),
+			'submit' => array(
+				'label'	=> 'SUBMIT',
+				'type'	=> 'submit',
+			),
+		);
+
+		if ($this->request->is_ajax())
+		{
+			$this->iohandler->add_user_form_group($form_title, $form_data);
+			$this->iohandler->send_response();
+		}
+		else
+		{
+			$rendered_form = $this->iohandler->generate_form_render_data($form_title, $form_data);
+
+			$this->template->assign_vars(array(
+				'TITLE'		=> $this->language->lang('STAGE_SETTINGS'),
+				'CONTENT'	=> $rendered_form,
+			));
+		}
+	}
+
+	/**
+	 * Render the list of available convertors
+	 *
+	 * @return \Symfony\Component\HttpFoundation\Response
+	 */
+	protected function render_convert_list()
+	{
+		$this->template->assign_vars(array(
+			'TITLE'		=> $this->language->lang('CONVERT_INTRO'),
+			'BODY'		=> $this->language->lang('CONVERT_INTRO_BODY'),
+			'S_LIST'	=> true,
+		));
+
+		$convertors = $sort = array();
+		$get_info = true; // Global flag
+
+		$handle = @opendir($this->phpbb_root_path . 'install/convertors/');
+
+		if (!$handle)
+		{
+			die('Unable to access the convertors directory');
```

Comment:
Thought it was a leftover and you wanted to change it into an exception or something like that.